MachObfuscator/Mach/ObjC/ObjcSymbols+Impl.swift

```diff
@@ -0,0 +1,342 @@
+//
+//  ObjcSymbols+Impl.swift
+//  MachObfuscator
+//
+
+import Foundation
+
+/// Architecture-erased protocol for 32/64 ObjC images
+private protocol ObjcImage {
+    func getObjectsOffsetsFromList(section list: Mach.Section?) -> [Int]
+    func getObjectsFromList<Element>(section list: Mach.Section?, creator: (Int) -> Element) -> [Element]
+    func create(offset: Int) -> ObjcClass
+    func create(offset: Int) -> ObjcCategory
+    func create(offset: Int) -> ObjcProtocol
+}
+
+/// Architecture-specific protocol for 32/64 ObjC images
+private protocol ObjcArchitecture {
+    // class_ro_t has padding that is present only in 32-bit version so we have to choose approriate struct.
+    // For other objc metadata generic 32/64-bit structs are enough.
+    associatedtype ClassData: ImageObjClass
+
+    typealias PointerType = ClassData.PointerType
+}
+
+private protocol ObjcArchitecture32: ObjcArchitecture where ClassData == ObjC.class_32 {}
+
+private protocol ObjcArchitecture64: ObjcArchitecture where ClassData == ObjC.class_64 {}
+
+private protocol FromMach {
+    init(mach: Mach, offset: Int)
+    /// Structure describing raw metadata structure in image
+    associatedtype Raw
+
+    var mach: Mach { get }
+    var offset: Int { get }
+}
+
+private extension FromMach {
+    var raw: Raw {
+        return mach.getStruct(atFileOffset: offset)
+    }
+}
+
+private struct ObjcMethodImpl<Arch: ObjcArchitecture>: ObjcMethod, FromMach, CustomStringConvertible {
+    typealias Raw = ObjC.method_t<Arch.PointerType>
+
+    fileprivate let mach: Mach
```

Comment:
`fileprivate` is redundant on `private` class properties. Could be dropped. Same for all other `fileprivate` properties in this file.

Reply:
making these `private` causes error: `Property 'mach' must be declared fileprivate because it matches a requirement in private protocol 'FromMach'`